test_main.py

```diff
@@ -22,7 +22,7 @@ def test_no_args(capsys):
 def test_len():
     # Database unfortunately makes db.__MAX_ID private :-(
     __MAX_ID = 493
-    assert len(db) == __MAX_ID + len(db.get_extra())
+    assert len(db.get_all()) == __MAX_ID + len(db.get_extra())
```

Comment:
Let's keep **both lines** here.  The red line tests the Database.__len__() and the red line tests db.get_all().  Both are valid and important tests.  The more tests helps find corner cases.